src/Netscript/APIWrapper.ts

```diff
@@ -0,0 +1,157 @@
+import { getRamCost } from "./RamCostGenerator";
+import type { IPort } from "../NetscriptPort";
+import type { BaseServer } from "../Server/BaseServer";
+import type { WorkerScript } from "./WorkerScript";
+import { makeRuntimeRejectMsg } from "../NetscriptEvaluator";
+import { Player } from "../Player";
+
+type ExternalFunction = (...args: any[]) => any;
+type ExternalAPI = {
+  [string: string]: ExternalAPI | ExternalFunction;
+};
+
+type InternalFunction<F extends (...args: unknown[]) => unknown> = (
+  ctx: NetscriptContext,
+) => (...args: unknown[]) => ReturnType<F>;
```

Comment:
nitpick `...args: Parameters<F>` is more robust I think.

Reply:
heck this entire declaration can be simplified as 
```ts
type InternalFunction<F extends (...args: unknown[]) => unknown> = (
  ctx: NetscriptContext,
) => F;
```

Reply:
adding a commit in a moment